RustFxA/PushNotificationSetup.swift

```diff
@@ -71,6 +71,22 @@ open class PushNotificationSetup {
         }
     }
 
+    public func updatePushRegistration(subscriptionResponse: SubscriptionResponse) {
+        let endpoint = subscriptionResponse.subscriptionInfo.endpoint
+        let publicKey = subscriptionResponse.subscriptionInfo.keys.p256dh
+        let authKey = subscriptionResponse.subscriptionInfo.keys.auth
+        let devicePush = DevicePushSubscription(endpoint: endpoint,
+                                                publicKey: publicKey,
+                                                authKey: authKey)
+        RustFirefoxAccounts.shared.accountManager.uponQueue(.main) { accountManager in
```

Comment:
This really needs to be on the main queue?

Reply:
This was a case of a blind copy - good catch, it doesn't need to be (in fact, the inner call will re-direct to a global queue 😭

The threading there needs some love but that's beyond the scope of this, for now, I'll switch this not to run it on the main queue